controlplane/kubeadm/controllers/controller.go

```diff
@@ -180,7 +180,7 @@ func (r *KubeadmControlPlaneReconciler) Reconcile(req ctrl.Request) (res ctrl.Re
 		if err := r.updateStatus(ctx, kcp, cluster); err != nil {
 			var connFailure *internal.RemoteClusterConnectionError
 			if errors.As(err, &connFailure) {
-				logger.Info("Could not connect to workload cluster to fetch status", "err", err)
+				logger.Info("Could not connect to workload cluster to fetch status", "err", err.Error())
```

Comment:
The logger should already have a way to call `.String` or get the information from the error. Have you tested that with this change the error actually shows up? That'd be a little strange.

Given that we're casting and checking the error above, we could do something like: 

```go
logger.Info("Could not connect to workload cluster to fetch status", "err", connFailure)
```

Reply:
Yes I tested it.

I don't see why `connFailure` would work any better - it doesn't support `.String()`.

I agree the logger should call `.Error()`, and asked in the description where that code lives.